docs/reference/contributing/guidelines/style.md

```diff
@@ -59,55 +59,57 @@ uint32_t adc_function(analogin_t *obj, uint32_t options)
 ```
 ##### Rules
 
-* Indentation - four spaces. Please do not use tabs.
+- Indentation - four spaces. Please do not use tabs.
 
-* Braces - K&R style.
+- Braces - K&R style.
 
-* One true brace style (1TBS) - use braces for statements of type `if`, `else`, `while` and `for` (exception [from K&R](http://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS)).
+- One true brace style (1TBS) - use braces for statements of type `if`, `else`, `while` and `for` (exception [from K&R](http://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS)).
 
-* One line per statement.
+- One line per statement.
 
-* Preprocessor macro starts at the beginning of a new line; the code inside is indented according to the code above it.
+- Preprocessor macro starts at the beginning of a new line; the code inside is indented according to the code above it.
 
-* Cases within `switch` are indented (exception from K&R).
+- Cases within `switch` are indented (exception from K&R).
 
-* Space after statements of type `if`, `while`, `for`, `switch`. The same applies to binary operators (like, `+` and `*`) and the ternary operator (`?` and `:`).
+- Space after statements of type `if`, `while`, `for`, `switch`. The same applies to binary operators (like, `+` and `*`) and the ternary operator (`?` and `:`).
 
-* Each line preferably has at most 120 characters.
+- Each line preferably has at most 120 characters.
 
-* Comments should use proper spelling and grammar.
+- Comments should use proper spelling and grammar.
 
-* For pointers, `*` is adjacent to a name (`analogin_t *obj`).
+- For pointers or references, the symbols `*` or `&` are adjacent to a name (`analogin_t *obj`. `analogin_t &obj`). If you omit the name, place the space between the type and `*` (such as `int *` or `int &`).
 
-* Don't leave trailing spaces at the end of lines.
+- For function return pointers or references, the symbols `*` or `&` are adjacent to a function name (`int *func()` or `int &func()`).
 
-* Empty lines should have no trailing spaces.
+- Don't leave trailing spaces at the end of lines.
 
-* Unix line endings are default option for files.
+- Empty lines should have no trailing spaces.
 
-* Use capital letters for macros.
+- Unix line endings are default option for files.
 
-* A file should have an empty line at the end.
+- Use capital letters for macros.
+
+- A file should have an empty line at the end.
```

Comment:
Hmm, do you mean "empty line at the end"? I would interpret that statement as meaning

    last line with content\n
    \n

Do you just mean "new line at the end"?

Reply:
This was just a formatting fix done by @AnotherButler . I haven't touched.

Yes, new empty line at the end of the file. Should we clarify this one or?

Reply:
Save it for a later tweak maybe. Nothing to do with this PR.

I'm assuming you mean you want this

    last line with content\n
    |    <- cursor position

as opposed to this:

    last line with content, unterminated|   <- cursor position

and you don't really want:

    last line with content\n
    \n     <- empty line
    |      <- cursor position